src/Controller/ModulesController.php

```diff
@@ -529,28 +532,35 @@ public function getThumbsUrls(array &$response) : void
     }
 
     /**
-     * Bulk change status for objects
+     * Bulk change actions for objects
      *
      * @return \Cake\Http\Response|null
      */
-    public function changeStatus() : ?Response
+    public function bulkActions() : ? Response
     {
+        $requestData = $this->request->getData();
         $this->request->allowMethod(['post']);
-        if (!empty($this->request->getData('ids') && is_string($this->request->getData('ids')))) {
-            $ids = $this->request->getData('ids');
-            $status = $this->request->getData('status');
+
+        if (!empty($requestData['ids'] && is_string($requestData['ids']))) {
+            $ids = $requestData['ids'];
+
+            // extract valid attributes to change
+            $attributes = array_filter(
+                $requestData['attributes'],
+                function ($value) {
+                    return ($value !== null && $value !== '');
+                }
+            );
+
             if (!empty($ids)) { // export selected (filter by id)
                 $ids = explode(',', $ids);
                 foreach ($ids as $id) {
-                    $data = [
-                        'id' => $id,
-                        'status' => $status,
-                    ];
+                    $data = array_merge($attributes, ['id' => $id]);
                     $this->apiClient->save($this->objectType, $data);
```

Comment:
A `try/catch` block is needed here, to display full error message without being redirected to a generic page.
Do't know if outside or inside foreach loop....

Reply:
i would do it inside and display which id had an error